workshop.pl

```diff
@@ -705,6 +707,47 @@ sub arg_handler {
     exit 15;
 }
 
+# copy over any files from any requirements
```

Comment:
I wonder if this should go into the same loop where we install the packages.  That way we can enforce ordering like we do for dependency installation.  That could come in handy if a eventually a build or manual command execution depends on one of these file copies.

Reply:
There was a reason I did this after but I forget why :) I'll have to refresh my memory on this.

Reply:
OK, it is more obvious why now.  You can't reference the source of the copy while in the chroot'd environment, right?  That's why I had it copy later once the chroot was done.

Reply:
Well that is annoying.  I think we could get around that but it would be messy.

One idea would involve using the directory handle that we are holding open to eventually escape the chroot and doing the copying through code as opposed to using a shell call.

Another idea is I could add a thread which stays outside of the chroot and we send commands to to perform actions.

Or we can just leave it like you did it.  Do you think the ordering will eventually matter?

I suppose at the very least we could move the copy to before package installation and you could copy to temporary locations and then when inside the chroot you could pull from there as needed.